src/java.base/share/classes/com/sun/crypto/provider/DHKEM.java

```diff
@@ -247,10 +249,17 @@ private byte[] DH(PrivateKey skE, PublicKey pkR)
 
         private byte[] ExtractAndExpand(byte[] dh, byte[] kem_context)
                 throws NoSuchAlgorithmException, InvalidKeyException {
-            HKDF kdf = new HKDF(hkdfAlgorithm);
-            SecretKey eae_prk = LabeledExtract(kdf, suiteId, null, EAE_PRK, dh);
-            return LabeledExpand(kdf, suiteId, eae_prk, SHARED_SECRET,
-                    kem_context, Nsecret);
+            KDF hkdf = KDF.getInstance(hkdfAlgorithm);
+            SecretKey eae_prk = LabeledExtract(hkdf, suiteId, EAE_PRK, dh);
+            try {
+                return LabeledExpand(hkdf, suiteId, eae_prk, SHARED_SECRET,
+                        kem_context, Nsecret);
+            } finally {
+                if (eae_prk instanceof SecretKeySpec s) {
+                    SharedSecrets.getJavaxCryptoSpecAccess()
+                            .clearSecretKeySpec(s);
```

Comment:
I wish we could use `s.destroy()` here instead.

Reply:
Yes, it'd be nice. I reopened https://bugs.openjdk.org/browse/JDK-8160206 and we can address this separately.

Reply:
Or in the meantime:

    } finally {
        // Best effort
        if (eae_prk instanceof SecretKeySpec) {
            SharedSecrets.getJavaxCryptoSpecAccess()
                    .clearSecretKeySpec(eae_prk);
        } else {
            try {
                eae_prk.destroy();
            } catch (DestroyFailedException e) {
                // swallow
            }
        }
    }

Reply:
Sounds good, thanks for the suggestion.

Reply:
This is what I did in my Exporter code.  Assuming you go in first, I'll update mine to use your Util method.